app.rb

```diff
@@ -88,4 +97,112 @@ def display_person(person)
   def display_books(book)
     puts "Title: \"#{book.title}\", Author: #{book.author}"
   end
+
+  # Method to save books to a JSON file
+  def save_books_to_json(filename)
+    File.open(filename, 'w') do |file|
+      json_data = @books.map(&:store_book)
+      file.write(JSON.pretty_generate(json_data))
+    end
+  end
+
+  def load_books_from_json(filename)
+    return [] unless File.exist?(filename)
+
+    json_data = JSON.parse(File.read(filename))
+    json_data.map do |book_data|
+      Book.new(book_data['title'], book_data['author'])
+    end
+  end
+
+  def save_people_to_json(filename)
+    File.open(filename, 'w') do |file|
+      json_data = @persons.map(&:json_data)
+      file.write(JSON.pretty_generate(json_data))
+    end
+  end
+
+  def load_people_from_json(filename)
+    return [] unless File.exist?(filename)
+
+    json_data = JSON.parse(File.read(filename))
+
+    json_data.map do |person_data|
+      build_person_from_data(person_data)
+    end.compact
+  end
+
+  def save_rentals_to_json(filename)
+    rentals_data = @rentals.map do |rental|
+      {
+        date: rental.date,
+        person_id: rental.person.id,
+        book_title: rental.book.title
+      }
+    end
```

Comment:
- [ ] According to the requirements you should only save the data when exiting the app not after each creation.

Reply:
noted

Reply:
when '7'
    app.save_books_to_json('local_db/books.json')
    app.save_people_to_json('local_db/people.json')
    app.save_rentals_to_json('local_db/rentals.json')
    print 'Thank you for using this app!'
    break
  end
  
  saving data while exiting app at main.rb